### DISCUSSION TITLE
Why not use  plz or request?

### DISCUSSION THREAD

**Discussion by tusharhero** [karthink/gptel#228] [Ideas]
Is there any particular reason for not using https://github.com/alphapapa/plz.el or tkf.github.io/emacs-request/ ?

**Comment by karthink** [comment]
Is there any particular reason to use them?

**Comment by tusharhero** [comment]
Reducing the amount of code you maintain and duplicating less can save time and effort.

**Comment by karthink** [comment]
I would still have to write the tiresome and finicky process sentinels and filters gptel-curl uses.  The only functions I could use plz for would be to replace gptel-curl-get-response and a bit of the error handling, and even these would look mostly the same because of the bookkeeping required on gptel's part.  Including the dependencies, the end result would be a big overall increase in the amount of code gptel uses.  Take a look at gptel-curl and let me know how specifically using plz or request can help.

**Comment by karthink** [comment]
Converting to a discussion since there isn't anything to fix/add at the moment.

**Comment by karthink** [comment]
Some more context: As it turns out you _cannot_ use `plz` or `request` for what gptel needs since these only provide for callbacks, and not filter functions.  This means streaming responses are not possible with `plz` or `request`.

Extensions for `plz` that add this feature are in the works (`plz-media-type` and `plz-event-source`), but these are unfinished and untested, and there's no reason to introduce them to gptel.
